### DISCUSSION TITLE
Spherical coordinates

### DISCUSSION THREAD

**Discussion by raphaelle98** [idaholab/moose#20690] [Q&A Meshing] [answered]
Hi everyone,

This is my first message here and I'm new in MOOSE so I'd like to say I'm glad to join this community!
I have a model where my mesh is in one dimension and I'm using the default cartesian coordinates. This runs well and converges very fast (1 minute). 
My goal is to run the same thing in one dimension but by using spherical (and also cylindrical) coordinates so I simulate a 3D (or 2D) with it. So I added a block [Problem] in my input file and specifies coord-type=RSPHERICAL (or RZ). 
And nothing converges anymore. How does this actually work and would you have any suggestion to make this converge? 
I have mesh adaptivity (uniform-refine=3) and also a time step adaptivity (with the tilmestep that can go until 10^-10, ...)

Thanks a lot for your help :)

**Comment by GiudGiud** [comment, marked as answer]
Hello

Switching to RSPHERICAL adapts the size of the elements in the background and also changes the definitions of common vector operators (grad etc) for that coordinate system. 

RSPHERICAL with R=0 can be challenging for some physics. Can you please quickly try to offset it by a little bit, translating the system to start at R=0.01 for example. 

Guillaume
